### PR TITLE
980 last accessed

### PR DESCRIPTION
**Pull request by kyhu65867**
# Committer Notes

Addresses issue 980. 

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
Separate of the file declarations, you may need to add the constraint ID (itself, `last-accessed-is-datetime`, observe not with the file names) and re-run. It is possible you are failing on coverage? You can see some examples lower in this file.

I am going to test this now locally.

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -740,4 +740,15 @@
         </constraints>
     </context>
 
+    <context>
+        <metapath target="//back-matter"/>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:744]: You need to adjust the context here or in the `matches/@target`, I would recommend here.

```suggestion
        <metapath target="//back-matter/resource"/>
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 2]
```diff
@@ -740,4 +740,15 @@
...
+        <constraints>
+            <matches id="last-accessed-is-datetime" target="prop[@ns='http://fedramp.gov/ns/oscal' and @name='last-accessed']/@value" datatype="dateTime-with-timezone" level="ERROR">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:746]: You can change the constraint above or here from `target="prop[@ns='http://fedramp.gov/ns/oscal' and @name='last-accessed']/@value"` to `target="resource/prop[@ns='http://fedramp.gov/ns/oscal' and @name='last-accessed']/@value"`, but this is not preferable to the `/context/metapath/target`, so I would do the above.

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 3]
```diff
@@ -740,4 +740,15 @@
         </constraints>
     </context>
 
+    <context>
+        <metapath target="//back-matter/resource"/>
+        <constraints>
+            <matches id="last-accessed-is-datetime" target="prop[@ns='http://fedramp.gov/ns/oscal' and @name='last-accessed']/@value" datatype="dateTime-with-timezone" level="ERROR">
+                <formal-name>Last Accessed is Type DateTime</formal-name>
+                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="add-help-url-here"/>
+                <message>FedRAMP Back Matter MUST specify the last accessed date as a "full-date" per RFC3339 with the addition of a timezone.</message>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:749]:
```suggestion
                <message>A FedRAMP document SHOULD specify the last accessed date for resources where the specification publication date is unknown.  In such cases, the last accessed date MUST be be a date with an explicit timezone, per RFC-3339.</message>
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
Minor wording change, and I am then completely happy with it until we sort out the docs.

**Review by Gabeblis — APPROVED** [review]

**Comment by Gabeblis** [comment]
Looks really good! I'll reapprove when the documentation is added 😄

**Comment by kyhu65867** [comment]
@Gabeblis @aj-stein-gsa I have added a documentation pull request here if you want to take a look: https://github.com/GSA/automate.fedramp.gov/pull/142/files

**Review by Gabeblis — APPROVED** [review]
LGTM

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**